packages/next/src/build/webpack/loaders/next-metadata-route-loader.ts

```diff
@@ -1,32 +1,35 @@
 import type webpack from 'webpack'
 import path from 'path'
-
-const staticFileRegex = /[\\/](robots\.txt|sitemap\.xml)/
-
-function isStaticRoute(resourcePath: string) {
-  return staticFileRegex.test(resourcePath)
-}
+import { stringifyRequest } from '../stringify-request'
+import { isStaticMetadataRoute } from '../../../lib/is-app-route-route'
 
 function getContentType(resourcePath: string) {
   const filename = path.basename(resourcePath)
-  const [name] = filename.split('.')
+  const [name, ext] = filename.split('.')
+  if (name === 'favicon' && ext === 'ico') return 'image/x-icon'
   if (name === 'sitemap') return 'application/xml'
   if (name === 'robots') return 'text/plain'
   return 'text/plain'
 }
 
-const nextMetadataRouterLoader: webpack.LoaderDefinitionFunction = function (
-  content: string
-) {
+// `import.meta.url` is the resource name of the current module.
+// When it's static route, it could be favicon.ico, sitemap.xml, robots.txt etc.
+const nextMetadataRouterLoader: webpack.LoaderDefinitionFunction = function () {
   const { resourcePath } = this
 
-  const code = isStaticRoute(resourcePath)
+  const code = isStaticMetadataRoute(resourcePath)
     ? `import { NextResponse } from 'next/server'
+import fs from 'fs'
+import path from 'path'
+import { fileURLToPath } from 'url'
+
+const filePath = ${stringifyRequest(this, resourcePath)}
+const relativePath = fileURLToPath(import.meta.url)
+const buffer = fs.readFileSync(relativePath)
```

Comment:
`filePath` is not needed right? Also, I'm not 100% sure if `@vercel/nft` catches this case so maybe you'll have to inline it:

```suggestion
const buffer = fs.readFileSync(fileURLToPath(import.meta.url))
```

Could you test it with `nft print test.js`?